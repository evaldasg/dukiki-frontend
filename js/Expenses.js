import React from 'react'
import NavBar from './NavBar'
import payload from '../data/expenses'

const Expenses = React.createClass({
  sortPayload (prop) {
    return payload.expenses.sort((first, second) => {
      return new Date(second[prop]) - new Date(first[prop])
    })
  },

  render () {
    console.log(this.props)
    let sortedPayload = this.sortPayload('date')
    console.log(sortedPayload)

    return (
      <div>
        <NavBar />
        <div className='container'>
          <div className='row'>
            <div className='col s9'>
              <h3 className='header orange-text'>Monthly Expenses</h3>
            </div>
            <div className='col s3'>
              <div className='expenses new-button'>
                <a className='expenses-new waves-effect waves-light btn-large right light-blue darken-2'>
                  <i className='material-icons left'>add_circle</i>New
                </a>
              </div>
            </div>
          </div>
          <table className='bordered striped highlight responsive-table'>
            <thead>
              <tr>
                <th>Date</th>
                <th>Description</th>
                <th>Category</th>
                <th>Amount</th>
              </tr>
            </thead>
            <tbody>
              {
                sortedPayload.map((element) => {
                  return (
                    <tr key={element.id}>
                      <td>{element.date}</td>
                      <td>{element.description}</td>
                      <td>{element.category.name}</td>
                      <td>{element.amount}</td>
                    </tr>
                  )
                })
              }
            </tbody>
          </table>
        </div>
      </div>
    )
  }
})

export default Expenses
