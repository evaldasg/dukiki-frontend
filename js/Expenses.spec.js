import React from 'react'
import Expenses from './Expenses'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

test('Expenses snapshot test', () => {
  const component = shallow(<Expenses />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})
