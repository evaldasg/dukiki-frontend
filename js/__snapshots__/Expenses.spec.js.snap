exports[`test Expenses snapshot test 1`] = `
<div>
  <NavBar />
  <div
    className="section no-pad-bot"
    id="index-banner">
    <div
      className="container">
      <br />
      <br />
      <h1
        className="header center orange-text">
        Expenses page
      </h1>
      <div
        className="row center">
        <h5
          className="header col s12 light">
          This is your personal best finance advisor app
          <br />
          in the world.
        </h5>
      </div>
      <div
        className="row center">
        <a
          className="btn-large waves-effect waves-light orange"
          href="#"
          id="download-button">
          Learn more...
        </a>
      </div>
    </div>
  </div>
</div>
`;
